src/ias_proxy/src/ias_proxy_server/mod.rs

```diff
@@ -0,0 +1,216 @@
+
+/// structure defining IAS proxy server
+pub struct IasProxyServer {
+    ias_proxy_name: String,
+    ias_proxy_port: String,
+    ias_client: IasClient,
+}
+
+/// response body from the IAS
+#[derive(Deserialize)]
+struct IasResponseBody {
+    #[serde(rename = "isvEnclaveQuote")]
+    isv_enclave_quote: String,
+    #[serde(rename = "pseManifest")]
+    pse_manifest: String,
+    nonce: u64,
+}
+
+lazy_static! {
+    static ref sig_rl_cache: Mutex<LruCache> = Mutex::new(LruCache::new(None));
+    static ref attestation_cache: Mutex<LruCache> = Mutex::new(LruCache::new(None));
+}
+
+static SIG_RL_LINK: &str = "/attestation/sgx/v2/sigrl";
+static AVR_LINK: &str = "/attestation/sgx/v2/report";
+
+impl IasProxyServer {
+    /// Create new instance of IasProxyServer
+    pub fn new(config_map: HashMap<String, String>) -> Self {
+        IasProxyServer {
+            ias_proxy_name: config_map["proxy_name"].clone(),
+            ias_proxy_port: config_map["proxy_port"].clone(),
+            ias_client: IasClient::new(
+                config_map["ias_url"].clone(),
+                Vec::from(config_map["spid_cert_file"].as_bytes()),
+                None,
+            ),
+        }
+    }
+
+    /// run method to start listeing on the identified port and IP
+    pub fn run(&self) -> () {
+        // Start the web server on the configured URL
+        let path = self.ias_proxy_name.as_str().to_owned() + ":" + &self.ias_proxy_port.as_str().to_owned();
+        info!("Path constructed is {}", path);
+        let socket_addr: SocketAddr = match SocketAddr::from_str(&path) {
+            Ok(found_addr) => found_addr,
+            Err(err) => panic!("Error reading the address: {}", err),
+        };
+        let ias_client = self.ias_client.clone();
+        // TODO: Store this server instance and call shutdown
+        // TODO: Solve below error (logic error) using mutex
+        let new_service = move || {
+            let ias_client = ias_client.clone();
+            service_fn(move |req|
+                respond_to_request(req, /*&sig_rl_cache, &attestation_cache, */&ias_client)
+            )
+        };
+        let server = Server::bind(&socket_addr).serve(new_service);
+        hyper::rt::run(server.map_err(|e| {
+            panic!("Server error: {}", e);
+        }));
+        // Note: This should also be doing a proxy job
+        // signed rl cache from the LruCache(), attestation cache from the LruCache()
+        // If get request when listening and it's for '/attestation/sgx/v2/sigrl' then get proxy
+        // If post request when listening and it's for '/attestation/sgx/v2/report' then post proxy
+        // Error case returns 404
+        // Get cached content if it's present otherwise send it to IasClient request for both GET and POST
+    }
+
+    /// stop to stop listening on the port
+    pub fn stop(&self) -> () {
+        // TODO: Need to stop the server started
```

Comment:
Missed some implementation or deliberate?

Reply:
It's deliberate, still couldn't figure out a way to stop the server. I was trying similar logic in unit test patch which is much smaller and simpler compared to this but wasn't successful. Will leave this as TODO until it's resolved. Problem is hyper doesn't have a working example which we want for the version we're using. Either information available online on hyper is outdated to older versions or there's no example available for the latest version. Available solutions didn't work for the version we're using. Tried few other alternate way of doing away but am not successful yet.